tasks/c++/hard/mystring.cpp

```diff
@@ -24,32 +46,78 @@ MyString::~MyString() {
 
 int MyString::length() const {
     // TODO: Implement this
+    return len;
 }
 
 const char* MyString::c_str() const {
     // TODO: Implement this
+    return str;
 }
 
 void MyString::append(const MyString& other) {
     // TODO: Implement this
+    len = (strlen(str) + strlen(other.str));
+    strcat(str, other.str);
+
 }
 
 int MyString::compare(const MyString& other) const {
     // TODO: Implement this
+    int com=0;
+    for (unsigned int i = 0; i <= len; ++i) {
+        if (str[i] < other.str[i]) {
+            com = -1;
+            break;
+        }
+        else if (str[i] > other.str[i]){
+            com = 1;
+            break;
+        }
+    }
+    return com;
 }
 
 MyString& MyString::operator=(const MyString& other) {
     // TODO: Implement this
+    if (this == &other) {
+        return *this;
+    }
+    delete[] str;
+    str = new char[strlen(other.str) + 1];
+    strcpy(str, other.str);
+    len = strlen(other.str);
+    return *this;
 }
 
 MyString MyString::operator+(const MyString& other) const {
     // TODO: Implement this
+    MyString temp;
+    temp.len = (len + other.len)+1;
+    temp.str = new char[temp.len];
+    strcpy(temp.str, str);
+    strcat(temp.str, other.str);
+    temp.str[len] = '\0';
+    return temp;
+
 }
```

Comment:
The class looks mostly good but it does not quite work. It prints only `Hello` when executing, and i think the problem might be not using proper memory allocation for temp.str in the operator+ method, which will cause it to crash. If you can fix that, i can merge this PR.